Wikipedia/Code/WMFTableOfContentsAnimator.swift

```diff
@@ -182,16 +189,16 @@ public class WMFTableOfContentsAnimator: UIPercentDrivenInteractiveTransition, U
             self.presentingViewController?.presentViewController(self.presentedViewController!, animated: true, completion: nil)
         case (.Changed):
             let position = gesture.locationInView(gesture.view);
-            let distanceFromRight = CGRectGetMaxX(gesture.view!.bounds) - position.x
-            let transitionProgress = distanceFromRight / CGRectGetMaxX(gesture.view!.bounds)
+            let distanceFromSide = WMFTableOfContentsAnimator.shouldMoveTOCToLeft() ? position.x : CGRectGetMaxX(gesture.view!.bounds) - position.x
+            let transitionProgress = distanceFromSide / CGRectGetMaxX(gesture.view!.bounds)
             self.updateInteractiveTransition(transitionProgress)
         case (.Ended):
             self.isInteractive = false
-            let velocityRequiredToPresent = -CGRectGetWidth(gesture.view!.bounds)
+            let velocityRequiredToPresent = -CGRectGetWidth(gesture.view!.bounds) * rtlMultiplier()
             let velocityRequiredToDismiss = CGRectGetWidth(gesture.view!.bounds)
             
             let velocityX = gesture.velocityInView(gesture.view).x
-            if velocityX > velocityRequiredToDismiss{
+            if velocityX * rtlMultiplier() > velocityRequiredToDismiss{
```

Comment:
can you double check this? if the velocity is -800 and the threshold is -500, the threshold _should_ be exceeded, but -800 is **less than** -500, not greater than